inst/tests/tests.Rraw

```diff
@@ -18059,6 +18059,13 @@ for (opt in c(0,Inf)) {
   testnum = 2233.44
 }
 options(old)
+# optimized := with gforce functions that can return lists #5403
+DT = data.table(grp=1:2, x=1:4)
+test(2233.45, DT[, `:=`(y = shift(x, type = "lag", n = 1), z = shift(x = x, type = "lead", n = 1)), by = grp], data.table(grp=1:2, x=1:4, y=c(NA,NA,1:2), z=c(3:4, NA, NA)))
+DT = data.table(grp = 1:2, x=1:4)
+test(2233.46, DT[, l := shift(x, n=c(0, 0)), by=grp], data.table(grp=1:2, x=1:4, l=lapply(1:4, rep, 2)))
```

Comment:
```suggestion
test(2233.46, DT[, l := shift(x, n=c(0, 0)), by=grp], data.table(grp=1:2, x=1:4, l=list(INT(1, 1), INT(2, 2), INT(3, 3), INT(4, 4))))
```

Reading I didn't correctly infer what the `lapply()` output would be, better to be explicit on the expected value. could also do `c(1L, 1L)` etc. to avoid `INT()`.